pkg/grpc/server.go

```diff
@@ -193,10 +193,22 @@ func (a *apiImpl) Stop() bool {
 	a.listenersLock.Lock()
 	defer a.listenersLock.Unlock()
 
+	log.Infof("Stopping %d listeners", len(a.listeners))
 	for _, listener := range a.listeners {
+		debugMsg := "unknown listener type: "
+		switch listener.srv.(type) {
+		case *grpc.Server:
+			debugMsg = "gRPC server listener: "
+		case *http.Server:
+			debugMsg = "http handler listener: "
+		}
 		if listener.stopper != nil {
+			debugMsg += "stopped"
 			listener.stopper()
+		} else {
+			debugMsg += fmt.Sprintf("not stopped in loop. Comparing with grpcServer pointer with listener.srv pointer (%p : %p)", a.grpcServer, listener.srv)
 		}
+		log.Info(debugMsg)
```

Comment:
Shall this be Info or rather debug?

Reply:
Ideally it should be debug but I don't know how to change the log level in just this test.

Reply:
Oh, I see. How about something like this:

```go
// testing.T implements this
type debugLogger interface {
	Logf(format string, args ...any)
}

// add field `dLog` to `apiImpl`
// pass `t.Logf`  to a constructor of `apiImpl` so that it is stored in `dLog`
// use `dLog` in the Stop func

func (a *apiImpl) Stop() bool {
	var logger debugLogger = log.Logger // .. the standard logger that we use normally
   	if a.dLog != nil {
		// Use customly-provided logger when set
		logger = dLog
	}
...
}
```

This is fully optional, but may be worth a try if we plan to run with this log output for a while.

Reply:
The problem with your suggestion is that our logger does not implement the interface:

```golang
type debugLogger interface {
	Logf(format string, args ...any)
}
```

But I've come up with a similar approach. Let me know what you think.